aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,495 @@
+package com.eharmony.aloha.models.multilabel
+
+import java.io.{Closeable, PrintWriter, StringWriter}
+
+import com.eharmony.aloha.audit.Auditor
+import com.eharmony.aloha.dataset.density.Sparse
+import com.eharmony.aloha.factory._
+import com.eharmony.aloha.factory.ri2jf.RefInfoToJsonFormat
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.models._
+import com.eharmony.aloha.models.multilabel.json.MultilabelModelReader
+import com.eharmony.aloha.models.reg.RegressionFeatures
+import com.eharmony.aloha.reflect.{RefInfo, RefInfoOps}
+import com.eharmony.aloha.semantics.Semantics
+import com.eharmony.aloha.semantics.func.{GenAggFunc, GenAggFuncAccessorProblems}
+import com.eharmony.aloha.util.{Logging, SerializabilityEvidence}
+import spray.json.{JsonFormat, JsonReader}
+
+import scala.collection.{immutable => sci, mutable => scm}
+import scala.util.{Failure, Success}
+
+
+// TODO: When adding label-dep features, a Seq[GenAggFunc[K, Sparse]] will be needed.
+// TODO: To create a Seq[GenAggFunc[K, Sparse]], a Semantics[K] needs to be derived from a Semantics[A].
+// TODO: MorphableSemantics provides this.  If K is *embedded inside* A, it should be possible in some cases.
+// TODO: An alternative is to pass a Map[K, Sparse], Map[K, Option[Sparse]], Map[K, Seq[Sparse]] or something.
+// TODO: Directly passing the map of LDFs avoids the need to derive a Semantics[K].  This is easier to code.
```

Comment:
I don't like the mention of LDF here.  That is a VW specific term and this class should not be VW specific.

Reply:
When we include TensorFlow support, for instance, it should hopefully be able to make use of this class.

Reply:
Do you believe we can jam TensorFlow into this?  Is it sufficiently general?